src/macaron/repo_finder/repo_finder_java.py

```diff
@@ -49,12 +46,20 @@ def find_repo(self, group: str, artifact: str, version: str) -> Iterator[str]:
         # - Try to extract SCM metadata and return URLs
         # - Try to extract parent information and change current artifact to it
         # - Repeat
-        group = group.replace(".", "/")
+        group = (purl.namespace or "").replace(".", "/")
```

Comment:
It may be better to do the `replace(".", "/")` in the `_create_urls` method, instead of doing it twice in this method (here, and in the `while` loop).